pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -447,10 +442,12 @@ public void publishMessage(ByteBuf headersAndPayload, PublishContext publishCont
 
     public void updateSubscribeRateLimiter() {
         SubscribeRate subscribeRate = this.getSubscribeRate();
-        if (isSubscribeRateEnabled(subscribeRate)) {
-            subscribeRateLimiter = Optional.of(subscribeRateLimiter.orElse(new SubscribeRateLimiter(this)));
-        } else {
-            subscribeRateLimiter = Optional.empty();
+        synchronized (subscribeRateLimiter) {
```

Comment:
remove `synchronized (subscribeRateLimiter) {`. That is invalid as I explained previously. Making the field subscribeRateLimiter volatile is the correct solution.

Reply:
OK. PTAL @lhotari :)